src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c

```diff
@@ -21,77 +46,699 @@ const ActorInit En_Goroiwa_InitVars = {
+            if (this->actor.xzDistFromLink < 600.0f) {
+                quakeIdx = Quake_Add(ACTIVE_CAM, 3);
+                Quake_SetSpeed(quakeIdx, -0x3CB0);
+                Quake_SetQuakeValues(quakeIdx, 3, 0, 0, 0);
+                Quake_SetCountdown(quakeIdx, 7);
+            }
+            this->unk_1C0 = 0.0f;
+            if (!(this->unk_1D3 & 0x10)) {
+                sp5C.x = this->actor.posRot.pos.x;
+                sp5C.y = this->actor.posRot.pos.y + 50.0f;
+                sp5C.z = this->actor.posRot.pos.z;
+                temp_f0_2 = func_8003CA0C(globalCtx, &globalCtx->colCtx, &sp68, &sp50, &this->actor, &sp5C);
+                // temp needed to match
+                temp = temp_f0_2 - (this->actor.posRot.pos.y - 59.5f);
+                if (fabsf(temp) < 15.0f) {
+                    sp44.x = this->actor.posRot.pos.x;
+                    sp44.y = temp_f0_2 + 10.0f;
+                    sp44.z = this->actor.posRot.pos.z;
+                    func_80A4C3A4(globalCtx, &sp44);
+                }
+            }
+        }
+        if (this->unk_1C6 > 0) {
+            return true;
+        }
+        this->unk_1C6 += 1;
+        this->actor.velocity.y *= -0.3f;
+        this->actor.posRot.pos.y = nextPointY - ((this->actor.posRot.pos.y - nextPointY) * 0.3f);
+    }
+    if (this->unk_1C6 == 0 &&
+        func_80042244(globalCtx, &globalCtx->colCtx, this->actor.posRot.pos.x, this->actor.posRot.pos.z, &ySurface,
+                      &waterBox) &&
+        this->actor.posRot.pos.y <= ySurface) {
+        this->unk_1D3 |= 0x10;
+        if (ySurface < thisY) {
+            sp30.x = this->actor.posRot.pos.x;
+            sp30.y = ySurface;
+            sp30.z = this->actor.posRot.pos.z;
+            func_80A4C594(globalCtx, &sp30);
+            this->actor.velocity.y *= 0.2f;
+        }
+        if (this->actor.velocity.y < -8.0f) {
+            this->actor.velocity.y = -8.0f;
+        }
+    }
+    return false;
+}
+
+void func_80A4CED8(EnGoroiwa* this, GlobalContext* globalCtx) {
+    static Vec3f unitY = { 0.0f, 1.0f, 0.0f };
+
+    s32 pad;
+    Vec3f* temp;
+    f32 sp8C;
+    Vec3f sp80;
+    Vec3f sp74;
+    MtxF mtx;
+    Vec3f sp28;
+
+    if (this->unk_1D3 & 8) {
+        sp8C = this->unk_1BC;
+    } else {
+        this->unk_1BC = Math3D_Vec3f_DistXYZ(&this->actor.posRot.pos, &this->actor.pos4) * 0.016806724f;
+        sp8C = this->unk_1BC;
+    }
+    sp8C *= this->unk_1C0;
+    // temp assignment needed to match
+    temp = &sp80;
+    if (this->unk_1D3 & 8) {
+        func_80A4BF28(this, globalCtx, &sp28);
+        Math3D_Vec3f_Cross(&unitY, &this->actor.velocity, temp);
+    } else {
+        Math3D_Vec3f_Cross(&unitY, &this->actor.velocity, temp);
+    }
+
+    if (EnGoroiwa_Vec3fNormalize(&sp74, temp)) {
+        this->unk_1B0 = sp74;
+    } else {
+        sp74 = this->unk_1B0;
+    }
+
+    func_800D23FC(sp8C, &sp74, 0);
+    Matrix_RotateY(this->actor.shape.rot.y * (2.0f * M_PI / 65536.0f), 1);
```

Comment:
does `0x10000` or `(f32)0x10000` work instead of `65536.0f`

Reply:
Both work. Would `0x10000` be preferred?

Reply:
I think so, because it's normalizing the s16 angle `this->actor.shape.rot.y` which we use hex for